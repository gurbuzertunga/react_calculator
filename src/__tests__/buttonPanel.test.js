import renderer from 'react-test-renderer';
import ButtonPanel from '../components/buttonPanel';

describe('checks rendering for ButtonPanel', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<ButtonPanel name="AC" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
