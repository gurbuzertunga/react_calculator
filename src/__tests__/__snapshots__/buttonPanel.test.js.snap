// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering for ButtonPanel should render correctly 1`] = `
<div
  className="button-container"
>
  <div
    className="top-buttons"
  >
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
  <div
    className="second-buttons"
  >
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="third-buttons"
  >
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="fourth-buttons"
  >
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="nums"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="fifth-buttons"
  >
    <button
      className="nums equal"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="basic-op"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
